### PR TITLE
feat(stepfunctions-tasks): start a nested state machine execution as a construct

### PR DESCRIPTION
**Pull request by shivlaks**
This class is the replacement for the previous `StartExecution` class.

There are a few differences:
1. the `stateMachine` parameter has been moved into props.
Rationale: alignment with constructs.

2. the resource ARN that's generated in the Amazon States
language uses `sync:2`. This returns an output of JSON instead
of a string.
Rationale: alignment with Step Functions team recommendation.

3. The `input` parameter has been changed to be of type `sfn.TaskInput`
Rationale: previous type precluded the ability to assign state input to
this parameter.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    this.taskPolicies = this.createScopedAccessPolicy();
+  }
+
+  protected renderTask(): any {
+    return {
+      Resource:
+        integrationResourceArn('states', 'startExecution', this.integrationPattern) +
+        (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB ? ':2' : ''),
```
Comment (jogold) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts:68]: using `.sync:2` is a good default, not sure an option for `.sync` is necessary.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  readonly stateMachine: sfn.IStateMachine;
+
+  /**
+   * The JSON input for the execution, same as that of StartExecution.
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html
+   *
+   * @default - No input
+   */
+  readonly input?: sfn.TaskInput;
```
[the remark this answers is not in the document]
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts:22]: I agree, flipping this to the state input. I think it's more convenient and often the intended input to the state machine (certainly more so than no input)

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/start-execution.ts [hunk 3]
```diff
@@ -39,6 +41,8 @@ export interface StartExecutionProps {
  * A Step Functions Task to call StartExecution on another state machine.
  *
  * It supports three service integration patterns: FIRE_AND_FORGET, SYNC and WAIT_FOR_TASK_TOKEN.
+ *
+ * @deprecated - use 'StepFunctionsStartExecution'
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/start-execution.ts:45]: Same comment as with batch. 

Can we call `new StepFunctionsStartExecution()` in the constructor here and delete most of the code in this file?

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts:68]: Add a comment here on what the `:2` is or a link to documentation.

Prefer using string interpolation over string concatenation using `+`.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/start-execution.ts (hunk 3 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/start-execution.ts:45]: I think that's pretty natural and organic if we were replacing one API with another to create an adapter and call into the replacement. I don't think this is a scenario where we can just proxy into the replacement as the replacement contains and configures properties (i.e. inputPath, outputPath, ...) that the older 

In this case, we are replacing a class that implemented a bind method to contribute a portion of the properties to represent a Task with a class that's a construct and configures all of the properties for a Task state.

The user experience with the current format will be to embed the call to start execution into the invocation of a new `Task`. I don't think proxying to invoke a new StepFunctionStartExeuction yields the right replacement.

```ts
new Task(this, 'my-task', {
  task: new tasks.StartExecution({...}),
...
}

Some of the other ones to come like DynamoDB and ECS, I've tried to reuse the shared types. I could perhaps move more into a central location here and DRY it up a bit.

Did you have something else in mind? If so, are you opposed to me making a subsequent PR to do some refactoring?
```

**Comment by shivlaks** [comment]
@nija-at would love if you could take another look when you get a chance!

**Pull request merged**